test/max-affected.ts

```diff
@@ -0,0 +1,138 @@
+import { InvalidMethodError } from '../src/PostgrestFilterBuilder'
+
+const REST_URL_13 = 'http://localhost:3001'
+const postgrest13 = new PostgrestClient<Database, { postgrestVersion: 13 }>(REST_URL_13)
+const postgrest12 = new PostgrestClient<Database>(REST_URL_13)
+
+describe('maxAffected', () => {
+  // Type checking tests
+  test('maxAffected should show type warning on postgrest 12 clients', async () => {
+    const resUpdate = await postgrest12
+      .from('messages')
+      .update({ channel_id: 2 })
+      .eq('message', 'foo')
+      .maxAffected(1)
+    expectType<InvalidMethodError<'maxAffected method only available on postgrest 13+'>>(resUpdate)
+  })
+  test('maxAffected should show type warning on non update / delete', async () => {
+    const resSelect = await postgrest13.from('messages').select('*').maxAffected(10)
+    const resInsert = await postgrest13
+      .from('messages')
+      .insert({ message: 'foo', username: 'supabot', channel_id: 1 })
+      .maxAffected(10)
+    const resUpsert = await postgrest13
+      .from('messages')
+      .upsert({ id: 3, message: 'foo', username: 'supabot', channel_id: 2 })
+      .maxAffected(10)
+    const resUpdate = await postgrest13
+      .from('messages')
+      .update({ channel_id: 2 })
+      .eq('message', 'foo')
+      .maxAffected(1)
+      .select()
+    const resDelete = await postgrest13
+      .from('messages')
+      .delete()
+      .eq('message', 'foo')
+      .maxAffected(1)
+      .select()
+    expectType<InvalidMethodError<'maxAffected method only available on update or delete'>>(
+      resSelect
+    )
+    expectType<InvalidMethodError<'maxAffected method only available on update or delete'>>(
+      resInsert
+    )
+    expectType<InvalidMethodError<'maxAffected method only available on update or delete'>>(
+      resUpsert
+    )
+    expectType<InvalidMethodError<'maxAffected method only available on update or delete'>>(
+      // @ts-expect-error update method shouldn't return an error
+      resUpdate
+    )
+    expectType<InvalidMethodError<'maxAffected method only available on update or delete'>>(
+      // @ts-expect-error delete method shouldn't return an error
+      resDelete
+    )
+  })
+
+  // Runtime behavior tests
+  test('update should fail when maxAffected is exceeded', async () => {
+    // First create multiple rows
+    await postgrest13.from('messages').insert([
+      { message: 'test1', username: 'supabot', channel_id: 1 },
+      { message: 'test1', username: 'supabot', channel_id: 1 },
+      { message: 'test1', username: 'supabot', channel_id: 1 },
+    ])
+
+    // Try to update all rows with maxAffected=2
+    const result = await postgrest13
+      .from('messages')
+      .update({ message: 'updated' })
+      .eq('message', 'test1')
+      .maxAffected(2)
+    const { error } = result
+    expect(error).toBeDefined()
+    expect(error?.message).toBe('Query result exceeds max-affected preference constraint')
```

Comment:
How about just matching the PostgREST error code? (`PGRST124`, [ref](https://docs.postgrest.org/en/v13/references/api/preferences.html#max-affected))

This would allow us to change the error message without causing a test failure later. Right now the error message seems a bit specific to postgREST ("preference constraint" could be removed)